### DISCUSSION TITLE
Is Supabase scalable for thousands of concurrent note-taking app users?

### DISCUSSION THREAD

**Discussion by eRuaro** [supabase/supabase#19944] [Questions]
Hi! 

I'm currently using Supabase to store notes created on my app. What I'm doing is that the notes are serialized into a JSON schema and every change to a note is an `update`  request to the DB on Supabase. I'm using plain React, and am using the `supabase-js` client for everything. Lately, I'm running into `max connections` reached error and am wondering on what's the strategy to go from here. I have over 100k users already, and average around 3k in daily active users. 

This means that multiple users are using it at the same time however, a portion of them can't do anything because of the `max connections` error which shows up as a notification to them. With that being said, I'm wondering about the approach I can do from here. Here are my options:

1. Migrate the notes table to a No-sql database
2. Instead of using `supabase-js`, I can create my own backend for connecting to the database to take advantage of a direct connection.
3. Instead of using plain React and `supabase-js`, I can migrate to Next.js, and use an ORM like Prisma to take advantage of using Supavisor. As far as I know, the JS client doesn't use Supavisor.

Would love to know your thoughts on what I can do here moving forward.

**Comment by MrVibe** [comment]
Are you using React-Query / Tanstack ? it will start caching results would help you reduce the max-connections.
Relational databases excel for complex data, like tables with joins which help generate statistics. If your app is simple then no-sql might be a good fit.

**Reply by eRuaro** [reply]
Only started looking into React-Query with supabase. As far as I know, it's kinda complicated to setup to work with `supabase-js`. Might be simpler to use React-Query with a custom backend that connects directly to the DB hosted through supabase.

**Reply by eRuaro** [reply]
I'm looking into migrating to no-sql as well - at the very least, the JSON schemas of the notes on the platform.

**Reply by GaryAustin1** [reply]
What size instance are you using for this load?

The REST API has its own pooler.   If you are exceeding connections you then likely your instance is not able to handle the load of queries you are throwing at it.

**Reply by eRuaro** [reply]
I'm on a medium instance, and supabase support upgraded the max connections to the postgREST pooler from 40, to 100.

**Comment by ambotwist** [comment]
Hello, I'm in the same boat. Did you find a solution in the end?
